src/gradualizer_lib.erl

```diff
@@ -222,6 +222,22 @@ pick_value(Type, Env)
             throw({undef, Kind, Anno, {Name, length(Args)}})
     end.
 
+pick_binary_value(?type(binary, []), _Env) ->
+    {bin, erl_anno:new(0), []};
+pick_binary_value(?type(binary, [{integer, _, 0}, {integer, _, _}]), _Env) ->
+    {bin, erl_anno:new(0), []};
+pick_binary_value(?type(binary, [{integer, _, M}, {integer, _, _}]), _Env) ->
+    Elements = case M rem 8 of
+                   0 ->
+                       S = [ $a || _ <- lists:seq(1, M div 8) ],
+                       [{bin_element, 0, {string, 0, S}, default, default}];
+                   _ ->
+                       Eights = [ {bin_element, 0, {integer, 0, 0}, default, default}
+                                  || _ <- lists:seq(1, M div 8) ],
+                       Tail = {bin_element, 0, {integer, 0, 0}, {integer, 0, M rem 8}, default},
+                       Eights ++ [Tail]
+               end,
+    {bin, erl_anno:new(0), Elements}.
```

Comment:
Interesting, but why not simply `<<0:M>>`?

Reply:
Good point. This mislead me:

```
12> <<1:48>>.
<<0,0,0,0,0,1>>
```

But indeed, it's just:

```
13> merl:quote("<<1:48>>").
{bin,1,
     [{bin_element,1,{integer,1,1},{integer,1,48},default}]}
```

I'll simplify it.